service-base/src/main/java/org/eclipse/hono/service/AbstractApplication.java

```diff
@@ -50,12 +48,12 @@
     protected final Logger log = LoggerFactory.getLogger(getClass());
 
     private ObjectFactory<S> serviceFactory;
-    private C config;
+    private ApplicationConfigProperties config;
```

Comment:
I think we should initalize this with a `new ApplicationConfigProperties()` to prevent *no null* checks ...